// agp-npm/dist/models/part.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.binPartSkinMap = exports.PartSkin = exports.PartType = void 0;
/** Represents each of an Axies body parts including: Eeyes, Ears, Mouth, Horn, Back, Tail. */
var PartType;
(function (PartType) {
    PartType["Eyes"] = "eyes";
    PartType["Ears"] = "ears";
    PartType["Mouth"] = "mouth";
    PartType["Horn"] = "horn";
    PartType["Back"] = "back";
    PartType["Tail"] = "tail";
})(PartType = exports.PartType || (exports.PartType = {}));
/** Represents the part skin of a given Axie. */
var PartSkin;
(function (PartSkin) {
    PartSkin["Global"] = "global";
    PartSkin["Mystic"] = "mystic";
    PartSkin["Japan"] = "japan";
    PartSkin["Xmas1"] = "xmas1";
    PartSkin["Xmas2"] = "xmas2";
    PartSkin["Bionic"] = "bionic";
})(PartSkin = exports.PartSkin || (exports.PartSkin = {}));
exports.binPartSkinMap = new Map([
    // 256 Classes
    ['00000', PartSkin.Global],
    ['00001', PartSkin.Japan],
    ['010101010101', PartSkin.Xmas1],
    ['01', PartSkin.Bionic],
    ['10', PartSkin.Xmas2],
    ['11', PartSkin.Mystic],
    // 512 PartSkins
    ['0000', PartSkin.Global],
    ['0001', PartSkin.Mystic],
    ['0011', PartSkin.Japan],
    ['0100', PartSkin.Xmas1],
    ['0101', PartSkin.Xmas2],
    ['0010', PartSkin.Bionic],
]);
